### ISSUE TITLE
Self-hosted server?

### ISSUE THREAD

**Issue by piegamesde** [laurent-clouet/sheepit-client#169]
Is there a way to self-host a sheepit server? I'd like to enable some features (OpenEXR images, scripts, etc.) for a personal project.

**Comment by MCOfficer** [comment]
Long story short: No.

**Comment by piegamesde** [comment]
That's a bummer. Could we maybe add an option for the client to allow blend files with OpenEXR output, for those with a better internet connection?

**Comment by MCOfficer** [comment]
That has been discussed a couple times, and even if 1/4th of the farm would render in exr, storing those frames would be very costly on the server. We usually have 2-3 GB free space, 

the user nubnubbud has done some (very rough) math on the topic:
> At the time of typing this message, there are 43 projects, out of 97, that can possibly be single frame renders (64 frames or fewer-this is the highest possible value for tiling a single image.)
> for the last 24 hours, the peak number of frames rendered was 4000, extrapolating to a day gets us 96000 frames a day
> combine the previous statistics, and we can make a generous assumption that half of the frames are single renders, leaving us with 48000 frames.
> (now here comes the iffy part)
> there are currently 324 frames, out of 77145 waiting, that can POSSIBLY belong to single frame renders!
> using the frame/day of 96000 and growing the single render frames(tiles) to match proportionally... we get 403 tiles rendered into about in a day, culminating in the completion of roughly 53 single frame renders!
> 
> alright, so how much will this impact something storing all those renders? let's take an incredibly noisy 8k HDR image (because noisy ones take up more data-(I tested this-in an 8k hdr image 3 samples made a 128mb image and 6 made a 124mb image)... and multiply that by both the tiles and the projects....
> [...]
> **in a single day, storing the individual tiles as .HDR files would take up 51.584 gigabytes(434.176 gigs if everyone used 64 tiles), and storing the finished projects would be 6.784 gigabytes**
> 
> **storing them as even noisier .EXR files (382mb each) would take up 153.946 gigabytes(1295.744 gigs if everyone used 64 tiles), and the finished projects would take up 20.246 gigabytes**

i didn't bother to double-check those numbers, but even if his estimation is double the actual number, and we disregard that this was in 2017, and the farm has grown since, **and** we assume only 1/4th of the farm is using .exr that's exactly the 2.5GB storage space we have currently available. 
but that's only for finished projects, the amount of storage the temporary tiles need dwarves this number.

bottom line - we'd need to invest in a much better server for exr support to happen.
